thorlcr/thorutil/thmem.cpp

```diff
@@ -286,12 +286,12 @@ class CSharedSpillableRowSet : public CSpillableStreamBase, implements IInterfac
         {
             pos = 0;
             outputOffset = (offset_t)-1;
-            owner->rows.registerWriteCallback(*this);
+            owner->rows.registerWriteCallback(*this); // NB: CStream construct called within rows lock
```

Comment:
This PR introduces safeRegisterWriteCallback + safeUnregisterWriteCallback (also introduced in HPCC-14083).
Here calling the non-safe version - as CStream constructor is called within a rows lock.